src/Agent.Listener/JobDispatcher.cs

```diff
@@ -632,17 +641,33 @@ private async Task RenewJobRequestAsync(int poolId, long requestId, Guid lockTok
                     }
                     else
                     {
-                        // retry till reach lockeduntil
-                        remainingTime = request.LockedUntil.Value - DateTime.UtcNow;
+                        // retry till reach lockeduntil + 5 mins oauth token time skew limit + 5 mins extra buffer for the delay of request monitor job on server side.
+                        remainingTime = request.LockedUntil.Value + TimeSpan.FromMinutes(5) + TimeSpan.FromMinutes(5) - DateTime.UtcNow;
                     }
 
                     if (remainingTime > TimeSpan.Zero)
                     {
-                        Trace.Info($"Retrying lock renewal for jobrequest {requestId}. Job is still locked for: {remainingTime.TotalSeconds} seconds.");
-                        TimeSpan delayTime = remainingTime.TotalSeconds > 60 ? TimeSpan.FromMinutes(1) : remainingTime;
+                        TimeSpan delayTime;
+                        if (!firstJobRequestRenewed.Task.IsCompleted)
+                        {
+                            Trace.Info($"Retrying lock renewal for jobrequest {requestId}. The first job renew request has failed.");
+                            delayTime = BackoffTimerHelper.GetRandomBackoff(TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(10));
+                        }
+                        else
+                        {
+                            Trace.Info($"Retrying lock renewal for jobrequest {requestId}. Job is valid until {request.LockedUntil.Value}.");
+                            delayTime = BackoffTimerHelper.GetRandomBackoff(TimeSpan.FromSeconds(5), TimeSpan.FromSeconds(15));
```

Comment:
Consider: is this too agressive for the entire duration of the remaining time? Have you considered changing `encounteringError` to Int32 and trying very agressively (5-15 sec) for the first 10 times, and then backing off to 15-30 seconds for the rest?